### PR TITLE
rustup 

### PR DESCRIPTION
**Pull request by matthiaskrgr**
changelog: none

### CODE REVIEW

**Comment by matthiaskrgr** [comment]
There's still this left
```
    Checking clippy_lints v0.0.212 (/home/matthias/vcs/github/rust-clippy/clippy_lints)
error[E0599]: no method named `is_local` found for enum `std::option::Option<rustc_span::def_id::DefId>` in the current scope
  --> clippy_lints/src/derive.rs:97:39
   |
97 |         if !&trait_ref.trait_def_id().is_local();
   |                                       ^^^^^^^^ method not found in `std::option::Option<rustc_span::def_id::DefId>`

error[E0277]: `rustc_hir::Path<'_>` doesn't implement `std::fmt::Display`
  --> clippy_lints/src/trait_bounds.rs:55:68
   |
55 | ...                   hint_string.push_str(&format!(" {} +", path));
   |                                                              ^^^^ `rustc_hir::Path<'_>` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `rustc_hir::Path<'_>`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: required because of the requirements on the impl of `std::fmt::Display` for `&rustc_hir::Path<'_>`
   = note: required because of the requirements on the impl of `std::fmt::Display` for `&&rustc_hir::Path<'_>`
   = note: required by `std::fmt::Display::fmt`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `rustc_hir::Path<'_>` doesn't implement `std::fmt::Display`
  --> clippy_lints/src/trait_bounds.rs:61:68
   |
61 | ...                   hint_string.push_str(&format!(" {} +", path));
   |                                                              ^^^^ `rustc_hir::Path<'_>` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `rustc_hir::Path<'_>`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: required because of the requirements on the impl of `std::fmt::Display` for `&rustc_hir::Path<'_>`
   = note: required because of the requirements on the impl of `std::fmt::Display` for `&&rustc_hir::Path<'_>`
   = note: required by `std::fmt::Display::fmt`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

```
I need to figure out which commit causes this (could be the rustc_hir_pretty one)

**Comment by flip1995** [comment]
Seems like paths has to be printed from their source snippets now: https://github.com/rust-lang/rust/pull/70344/files#diff-f7c5ab0a133d5d59a6ce76d8c88ff203R1316-R1320

**Comment by matthiaskrgr** [comment]
Yeah, it should work now :)

**Comment by matthiaskrgr** [comment]
Tests are failing :/

**Comment by flip1995** [comment]
Yeah, `Debug` is now derived for `hir::Path`, so I guess the span and res will be printed alongside the path itself?

You can resolve it like it was done in rustc: https://github.com/Centril/rust/blob/b2f7a9502bafebfa9ce0b61ff24a19c26e66c93b/src/librustc_privacy/lib.rs#L1316-L1321

and use `snippet(path.span)` to print the path

File: clippy_lints/src/derive.rs [hunk 1]
```diff
@@ -94,7 +94,7 @@ fn check_hash_peq<'a, 'tcx>(
     if_chain! {
         if match_path(&trait_ref.path, &paths::HASH);
         if let Some(peq_trait_def_id) = cx.tcx.lang_items().eq_trait();
-        if !&trait_ref.trait_def_id().is_local();
+       if !&trait_ref.trait_def_id().map_or(false, |def_id| def_id.is_local());
```
Comment (flip1995) [inline clippy_lints/src/derive.rs]: formatting. But while you're at it, can you change it back to the previous version, where `if let Some(def_id) ...` and `!def_id.is_local()` was separated. I liked that better than my suggestion.

Comment (matthiaskrgr) [reply clippy_lints/src/derive.rs]: Hm, apparently rustfmt doesn't like `if_chain!` :/

Comment (flip1995) [reply clippy_lints/src/derive.rs]: Yep, rustfmt only works in specific/common macros

**Comment by Centril** [comment]
I would recommend trying to avoid any further dependence on HIR pretty printing and trying to remove that which you already have. It's something we might want to remove and its not really a good fit for diagnostics in either case.

**Pull request merged**